### ISSUE TITLE
Multiple variation images a default feature

### ISSUE THREAD

**Issue by manake** [woocommerce/woocommerce#21077]
**Proposal:**
https://woocommerce.com/products/woocommerce-additional-variation-images/ should be a default WC feature.

**Justification:**
1. For variable products it's a very common use case that someone would have e.g. "4 red photos" and "4 blue photos" of the same item.
2. Currently it's only possible to upload 1 variation photo (e.g "blue"). On selecting "blue" color the product has "1 blue photo + 3 red photos" (makes no sense).
3. If I have "Color" and "Size" attributes and if I select just "Color" to "blue" then the 1st photo is not updated (that's a bug). Only after selecting the "Size" the photo is updated to "blue".

My config:
- 2 variations
- "Color: Red" + "Any size..."
- "Color: Blue" + "Any size..." (alternatively "Any color..." + "Any size...")

**Comment by woocommercebot** [comment]
Thanks for the suggestion @manake.

We'll keep an eye on the [popularity of this request](https://github.com/woocommerce/woocommerce/blob/master/.github/CONTRIBUTING.md#feature-requests) :)

**Comment by pacmanito** [comment]
I would like to second this. Now in 2021 multiple photos per variation is a must. Of course, now there are plenty of solutions - there are free and paid plugins, and also many paid themes implemented this in some way.
But there's a big problem with this approach - you can't integrate your store properly with other platforms and apps (e.g. marketplaces, inventory management solutions).
I can give you 3 examples:
1) you sell your products on many storefronts including EBay, Amazon, and WooCommerce store. To sync inventory and manage products you use some inventory management solution where you enter all the data (description, price, photos) for every product, and then it is listed to the channels. Currently, you will get only the first variation image uploaded to WooCommerce as there's no standard implementation for multiple variation images. You can imagine how much time can be spent uploading these variation images manually (especially due to the fact that there's no option to exclude some attributes from affecting images now and you have to upload images to every variation even if it just different size - https://github.com/woocommerce/woocommerce/issues/29033)
2) you manage your products through WooCommerce and use a product feed to send your products to some other platforms (e.g. Facebook). I bet you will have to spend a lot of time searching for a solution to add your custom implemented additional variation images to this feed.
3) you just want to integrate your WooCommerce with whatever service through REST API and again you can't manipulate your additional variation images because there's no standard REST API endpoint.

The plugin is 7 years old now and a lot of things have changed since it was released in 2014. Now in 2021 additional product variation images is a must, so please include this plugin in the core. The problem is not that we don't want to pay $49, the problem is that there will be no standard implementation until this plugin is merged into core and until there is no standard implementation all sorts of integrations are much more troublesome than they could be.

**Comment by pacmanito** [comment]
While maintaining my shop I met another scenario where core variations images would make much sense. Now when you delete a product all product images stay in media and should be deleted manually. This is another routine that should be automated (there's a request - https://github.com/woocommerce/woocommerce/issues/30870) by default, but we can achieve this at least through plugins now. But only for simple products or products without additional variation images, as soon as you implement some custom way of adding extra variation images you have to delete them manually.
